File: frontend/src/services/EmployeeService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/employees";

export const getEmployees = () => {
    return axios.get(API_URL);
};

export const addEmployee = (employee) => {
    return axios.post(API_URL, employee);
};

export const getEmployeeById = (id) => {
    return axios.get(`${API_URL}/${id}`);
};

export const updateEmployee = (id, employee) => {
    return axios.put(`${API_URL}/${id}`, employee);
};

export const deleteEmployee = (id) => {
    return axios.delete(`${API_URL}/${id}`);
};

File: backend/src/main/java/com/example/Employee_Management_System/service/EmployeeService.java
package com.example.Employee_Management_System.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.Employee_Management_System.entity.Employee;
import com.example.Employee_Management_System.repository.EmployeeRepository;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository employeeRepository;

    // Get all employees
    public List<Employee> getAllEmployees() {
        return employeeRepository.findAll();
    }

    // Get employee by id
    public Optional<Employee> getEmployeeById(Long id) {
        return employeeRepository.findById(id);
    }

    // Add employee
    public Employee addEmployee(Employee employee) {
        return employeeRepository.save(employee);
    }

    // Update employee
    public Employee updateEmployee(Long id, Employee updatedEmployee) {

        Employee employee = employeeRepository.findById(id).orElse(null);

        if (employee != null) {
            employee.setName(updatedEmployee.getName());
            employee.setEmail(updatedEmployee.getEmail());
            employee.setDepartment(updatedEmployee.getDepartment());
            employee.setSalary(updatedEmployee.getSalary());
            employee.setRole(updatedEmployee.getRole());

            return employeeRepository.save(employee);
        }

        return null;
    }

    // Delete employee
    public String deleteEmployee(Long id) {

        employeeRepository.deleteById(id);

        return "Employee deleted successfully";
    }
}